fix/to use pull_request_target

yash-zededa opened the pull request:
The PR builds are being executed without Docker login, leading to issues with rate limiting. Since `build.yml` cannot access the repository secrets with the `pull_request` event, we need to switch to `pull_request_target`. This will allow forked PRs to access the repository secrets, enabling the PR builds to pull Docker images within the authenticated user rate limit.

However, using `pull_request_target` has its drawbacks. Any changes to the workflow could potentially expose the secrets.. Therefore, to mitigate the risk of exposing secrets, we will not build PRs that include changes to the workflow files.

The above solution is interim. An ideal solution would be to check if there are any changes in the workflow files. If there are, we should request approval from a list of reviewers. Once approved, we can proceed to build the PR.


I will be testing this approach on sample repositories to verify the GitHub Actions workflow. Once it is confirmed to be working, I will raise a PR for the implementation.

deitch commented:
I saw this and was going to review it, and then saw it was draft. I think it makes sense, although, as you wrote, we will need to be careful. The approach looks good. Now that it is a regular PR, all good.

deitch commented:
Any idea why the git fetch is failing? Is it something to do with the the context change due to the changed event to `pull_request_target`?

milan-zededa commented:
@yash-zededa Could you please also add this (+ docker secrets) to eden.yml?

yash-zededa commented:
> I saw this and was going to review it, and then saw it was draft. I think it makes sense, although, as you wrote, we will need to be careful. The approach looks good. Now that it is a regular PR, all good.

@deitch I don't think so this is because of the context changed. We had the same issue with [9773306735](https://github.com/lf-edge/eve/actions/runs/9773306735).

I think the issue is because 

> we are attempting to fetch changes directly into the branch that is currently checked out, which Git does not allow to avoid potential conflicts and overwrites.

Considering we are only modifying the WF we can ignore for now.

yash-zededa commented:
> @yash-zededa Could you please also add this (+ docker secrets) to eden.yml?

@milan-zededa  I reviewed the changes @uncleDecart made for the Docker secrets and noticed some inconsistencies with the Eden version. I've updated the version to `0.9.9` and added the parameter `secrets: inherit`, which should ideally pass the secrets from the calling workflow repository to the called workflow (Eden in our case).

That said, we need to test the workflow. Once the PR is approved, we should be able to see the results.

milan-zededa commented:
@yash-zededa Seems that build workflows are not running for this PR (and eden tests depend on them).

milan-zededa commented:
Let's this...
